Decant `README` into Topiary Book

Xophmeister opened the pull request:
Decant the original `README.md` into the Topiary Book, making adjustments as necessary.

> [!NOTE]
> This PR is only for decanting and **minor** adjustments, to speed the process along. More significant changes to the text can be done in downstream PRs. (Hence, expect a lot of "TODOs" and "FIXMEs".)
>
> It is meant to be rough :sweat_smile:

yannham reviewed (commented):
A couple points on the overall approach:

1. Even if we have a nice Mdbook, we should keep a skeletal README with basic information. Other projects (rustc, LALRPOP, etc.) do as well, since this is the landing page of the repo. At least a generic description should stay and a link to the mdbook.
2. I would keep the whole README during the transition, precisely because we don't have a nice book to link to yet, so the README is still the one and only source of technical documentation for users currently. We can always scrap it at the end.

Xophmeister commented:
> A couple points on the overall approach:
> 
> 1. Even if we have a nice Mdbook, we should keep a skeletal README with basic information. Other projects (rustc, LALRPOP, etc.) do as well, since this is the landing page of the repo. At least a generic description should stay and a link to the mdbook.
> 
> 2. I would keep the whole README during the transition, precisely because we don't have a nice book to link to yet, so the README is still the one and only source of technical documentation for users currently. We can always scrap it at the end.

Of yes, of course, the `README.md` will not disappear. I just removed content from it to go into the book so I could track progress. In the super-PR (#884), there's a note to reinstate some kind of `README`, and I had in mind exactly as you suggest.

That is, I don't plan on merging all this into `main`. I've just split the PR into sub-PRs, to maintain momentum. The sub-PRs will merge into the super-PR and, when that's all ready, then it'll get merged into `main`.
